ui/src/app/shared/pod-name.ts

```diff
@@ -19,7 +19,8 @@ export function getPodName(workflow: Workflow, node: NodeStatus): string {
     }
 
     const workflowName = workflow.metadata.name;
-    if (workflowName === node.name) {
+    const podNodeName = node.type == 'Container' ? node.name.replace(/\.[^/.]+$/, '') : node.name;
```

Comment:
hmm this doesn't seem to match [the back-end](https://github.com/argoproj/argo-workflows/blob/0c1398ea6fa66cc77156f15a6d69c260c6b524da/workflow/util/pod_name.go#L51). these two functions should be equivalent per the comment on line 14 in this file

Reply:
Sure, reverted the change and moved it outside the function call. Maybe there are still better ways to do this

Reply:
It would be great if you could locate where in the back-end this gets set, that way we can keep them equivalent, whether in this function or elsewhere.

Reply:
Looks like the pod is created based on the pod node name.
https://github.com/argoproj/argo-workflows/blob/main/workflow/controller/container_set_template.go#L20-L24

Then the containerset nodes are created
https://github.com/argoproj/argo-workflows/blob/main/workflow/controller/container_set_template.go#L31-L46
Where the container node names are just their pod node name concatenated with the container name.
```go
ctxNodeName := fmt.Sprintf("%s.%s", nodeName, c.Name)
```
So removing this postfix should map from containerset node name to pod node name, which can be used to calculate the pod name using the existing function.

Reply:
Yea it looks like ContainerSet node -> Pod node name never happens directly in the back-end, which explains why they're not equivalent.

In that case, I'd prefer your earlier variant, since I intentionally consolidated `getPodName` in #12964.
Let's just leave a comment on why this block differs from the back-end then for explicitness

Reply:
Reverted it back to the old variant and added some comments in https://github.com/argoproj/argo-workflows/pull/13039/commits/39c44cb3e7c78a43fa47413f0f94a910dfa7d935

GH PRs still don't record _all_ versions of amended commits (unlike Gerrit which has had this for at least a decade if not more 😕), but I remembered the new GH activity feed existed and was able to [find it there](https://github.com/instauro/argo-workflows/activity?ref=fix/containerset-podname).